src/librustc_middle/mir/interpret/value.rs

```diff
@@ -60,6 +60,18 @@ impl<'tcx> ConstValue<'tcx> {
         self.try_to_scalar()?.to_bits(size).ok()
     }
 
+    pub fn try_to_bool(&self) -> Option<bool> {
+        match self.try_to_bits(Size::from_bytes(1))? {
+            0 => Some(false),
+            1 => Some(true),
+            _ => None,
+        }
+    }
+
+    pub fn try_to_usize(&self, tcx: TyCtxt<'tcx>) -> Option<u64> {
```

Comment:
Elsewhere in Miri we'd call this `try_to_machine_usize`; would that name also make sense for ConstValue?

Reply:
jop, I'll do that in a separate commit